site/layouts/partials/nav.html

```diff
@@ -1,17 +1,19 @@
 {{ $currentPage := . }}
-{{ $course_id := .CurrentSection.Params.course_id }}
-{{ $menu := index .Site.Menus $course_id }}
-<nav class="course-nav">
-  <ul class="m-auto">
-    {{ range $menu }}
-    <li class="pr-3">
-      <a
-        class="text-uppercase text-dark{{ if $currentPage.IsMenuCurrent $course_id . }} active{{ end }}"
-        href="{{ .URL | relURL }}"
-        >
-        {{ .Name }}
-      </a>
-    </li>
-    {{ end }}
-  </ul>
-</nav>
+{{ if (isset .CurrentSection.Params "course_id") }}
```

Comment:
in what cases will this not be set?

Reply:
Before adding hierarchical content support, the only "sections" we had were the courses themselves.  Thus, `.CurrentSection` would always refer to the course home page.  With the hierarchy now being dynamically constructed from the parent-child relationship in OCW data, new "sections" are being created.  I'm realizing now that as a stopgap measure so I could continue working on it, I just set `course_id` on every markdown file.  We can probably be more efficient about it though and explicitly grab the course home page in this template using `.Site.GetPage` somehow.

Reply:
idk, if we have `course_id` everywhere that seems like an easy solution to me - it means that wherever we are we always have a UUID for the course right handy and that lets us easily look up what we need. it's not like it costs us anything to just put it there, so why not?

Reply:
I suppose I was thinking about the files taking up extra space, but it doesn't really matter because it's a quite inconsequential amount of data.  At one point I was developing a web app to be served from a device with 25MB of total storage space... I still have an instinct to save a few kilobytes here and there where I can.

Reply:
not a bad instinct in general I think!